src/main/java/de/tum/in/www1/artemis/service/TitleCacheEvictionService.java

```diff
@@ -100,6 +102,16 @@ else if (entity instanceof ExerciseHint hint) {
             var combinedId = hint.getExercise().getId() + "-" + hint.getId();
             evictIdFromCache("exerciseHintTitle", combinedId);
         }
+        else if (entity instanceof ExerciseGroup exerciseGroup) {
+            if (!Hibernate.isInitialized(exerciseGroup.getExercises())) {
+                log.warn("Unable to clear title of exercises from exercise group {}: Exercises not initialized", exerciseGroup.getId());
+                return;
+            }
+            var exercises = exerciseGroup.getExercises();
+            for (Exercise exercise : exercises) {
+                evictIdFromCache("exerciseTitle", exercise.getId());
```

Comment:
why is this necessary?

Reply:
We should remove the title from the cache if the title changes to not get outdated cache values.